generate_workspace/src/main/java/com/google/devtools/build/workspace/output/BzlWriter.java

```diff
@@ -41,7 +43,9 @@ public BzlWriter(String[] argv, String outputDirStr) {
 
   @Override
   public void write(Collection<Rule> rules) {
-    try (PrintStream outputStream = new PrintStream(generatedFile.toFile())) {
+    try {
```

Comment:
Yeah, because it's different errors (creating directories vs. streaming content) and this way you don't have to worry about closing the output stream on failure.